### PR TITLE
Rust 1.0pre compat

### PR DESCRIPTION
**Pull request by johnpmayer**
haven't seen this started yet, trying to capture the low-hanging fruit

this isn't finished

### CODE REVIEW

File: codegen/branchify.rs [hunk 1]
```diff
@@ -104,13 +104,13 @@ pub fn generate_branchified_method(
             let next_prefix = format!("{}{}", prefix, c as char);
             w!(format!("Ok(b'{}') => match {} {{", c as char, read_call));
             for b in branch.children.iter() {
-                try!(r(writer, b, next_prefix[], indent + 1, read_call, end, max_len, valid, unknown));
+                try!(r(writer, b, next_prefix.as_slice(), indent + 1, read_call, end, max_len, valid, unknown));
```
Comment (chris-morgan) [inline codegen/branchify.rs]: The better transformation here is `&next_prefix[]`.

File: codegen/main.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#![feature(macro_rules, slicing_syntax)]
+#![feature(slicing_syntax, box_syntax, int_uint)]
```
Comment (chris-morgan) [inline codegen/main.rs]: It'd be better to update these as appropriate rather than use the gates.

File: codegen/main.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ fn main() {
     Thread::spawn(move || {
         let output_dir = Path::new(os::getenv("OUT_DIR").unwrap());
         read_method::generate(output_dir).unwrap();
-    }).detach();
+    });
```
Comment (chris-morgan) [inline codegen/main.rs]: ?

I think what is actually desired here is to `.join()` the guard at the end of the function. Or just ditch the threading, it's probably a silly idea anyway.

Comment (reem) [inline codegen/main.rs]: Yes, right now this will totally detach the thread and can leave things undone when main exits and all threads are killed. Either we should use `let join = Thread::scope(..)` and let `join` drop at the end of main, or we should just drop the threading altogether.

File: src/http/lib.rs [hunk 4]
```diff
@@ -7,14 +7,9 @@
 
 #![allow(unknown_features)]
 #![feature(slicing_syntax)]
-#![feature(default_type_params)]
-#![feature(macro_rules)]
-#![feature(phase)]
-#![feature(globs)]
-#![feature(associated_types)]
-#![feature(old_orphan_check)]
+//#![feature(old_orphan_check)]
```
Comment (reem) [inline src/http/lib.rs]: Should be actually deleted.
